rpcs3/rpcs3qt/log_frame.cpp

```diff
@@ -343,8 +340,11 @@ void log_frame::CreateAndConnectActions()
 	{
 		std::string text = m_tty_input->text().toStdString();
 
+		const auto tty = g_fxo->get<tty_t>();
+
+		if (tty)
```

Comment:
why do you check for tty here, but not in the rest of the `g_fxo->get<tty_t>()` uses ?

Reply:
I assume sys_tty syscalls happen only when tge emulation is running and tty is not nullptr, am I missing something?

Reply:
it's not transparent. And also not future proof.

Reply:
Explain why it is not future proof.

Reply:
I'm not gonna list future timelines and their influence on our code.
Maybe just assert it or whatever.

Anyway. Since this function is only called once in rpcs3 lifetime the tty check must never fail. So having it here is misleading in a sense as well.

Reply:
If emulation is not running its nullptr, simple as that.